diff --git a/src/modules/xwindow.cpp b/src/modules/xwindow.cpp
index 4afde7d8..02259faf 100644
--- a/src/modules/xwindow.cpp
+++ b/src/modules/xwindow.cpp
@@ -68,11 +68,6 @@ namespace modules {
       throw module_error("Failed to initialize ewmh atoms");
     }
 
-    // Check if the WM supports _NET_ACTIVE_WINDOW
-    if (!ewmh_util::supports(_NET_ACTIVE_WINDOW)) {
-      throw module_error("The WM does not list _NET_ACTIVE_WINDOW as a supported hint");
-    }
-
     // Add formats and elements
     m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL});
 
diff --git a/src/modules/xworkspaces.cpp b/src/modules/xworkspaces.cpp
index e6c3cee4..d07a1536 100644
--- a/src/modules/xworkspaces.cpp
+++ b/src/modules/xworkspaces.cpp
@@ -50,16 +50,6 @@ namespace modules {
       throw module_error("Failed to initialize ewmh atoms");
     }
 
-    // Check if the WM supports _NET_CURRENT_DESKTOP
-    if (!ewmh_util::supports(m_ewmh->_NET_CURRENT_DESKTOP)) {
-      throw module_error("The WM does not support _NET_CURRENT_DESKTOP, aborting...");
-    }
-
-    // Check if the WM supports _NET_DESKTOP_VIEWPORT
-    if (!(m_monitorsupport = ewmh_util::supports(m_ewmh->_NET_DESKTOP_VIEWPORT)) && m_pinworkspaces) {
-      throw module_error("The WM does not support _NET_DESKTOP_VIEWPORT (required when `pin-workspaces = true`)");
-    }
-
     // Add formats and elements
     m_formatter->add(DEFAULT_FORMAT, TAG_LABEL_STATE, {TAG_LABEL_STATE, TAG_LABEL_MONITOR});
 
@@ -106,11 +96,6 @@ namespace modules {
 
   void xworkspaces_module::update_current_desktop() {
     m_current_desktop = ewmh_util::get_current_desktop();
-    if (m_current_desktop < m_desktop_names.size()) {
-      m_current_desktop_name = m_desktop_names[m_current_desktop];
-    } else {
-      throw module_error("The current desktop is outside of the number of desktops reported by the WM");
-    }
   }
 
   /**
