diff --git a/src/modules/xwindow.cpp b/src/modules/xwindow.cpp
index 4afde7d8..02259faf 100644
--- a/src/modules/xwindow.cpp
+++ b/src/modules/xwindow.cpp
@@ -67,11 +67,6 @@ namespace modules {
       throw module_error("Failed to initialize ewmh atoms");
     }
 
-    // Check if the WM supports _NET_ACTIVE_WINDOW
-    if (!ewmh_util::supports(_NET_ACTIVE_WINDOW)) {
-      throw module_error("The WM does not list _NET_ACTIVE_WINDOW as a supported hint");
-    }
-
     // Add formats and elements
     m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL});
